packager/media/formats/mp2t/ts_segmenter.cc

```diff
@@ -171,30 +171,46 @@ Status TsSegmenter::WritePesPacketsToFile() {
 Status TsSegmenter::FinalizeSegment(uint64_t start_timestamp,
                                     uint64_t duration) {
   if (!pes_packet_generator_->Flush()) {
-    return Status(error::MUXER_FAILURE,
-                  "Failed to flush PesPacketGenerator.");
+    return Status(error::MUXER_FAILURE, "Failed to flush PesPacketGenerator.");
   }
   Status status = WritePesPacketsToFile();
   if (!status.ok())
     return status;
 
-  // This method may be called from Finalize() so ts_writer_file_opened_ could
+  // This method may be called from Finalize() so start_of_new_segment_ could
   // be false.
-  if (ts_writer_file_opened_) {
-    if (!ts_writer_->FinalizeSegment()) {
-      return Status(error::MUXER_FAILURE, "Failed to finalize TsWriter.");
+  if (start_of_new_segment_) {
+    std::string segment_path =
+        GetSegmentName(muxer_options_.segment_template, segment_start_timestamp_,
+                       segment_number_++, muxer_options_.bandwidth);
+
+  if ( segment_buffer_.Size() != 0) {
+
+    std::unique_ptr<File, FileCloser> segment_file;	  
+    segment_file.reset(File::Open(segment_path.c_str(), "w"));
+    if (!segment_file) {
+      LOG(ERROR) << "Failed to open file " << segment_path;
     }
+  
+  segment_buffer_.WriteToFile(segment_file.get()).ok();
+
+  if (!segment_file.release()->Close()) {
+    return Status(error::MUXER_FAILURE, "Failed to finalize TsWriter.");
```

Comment:
return Status(
        error::FILE_FAILURE,
        "Cannot close file " + segment_path +
            ", possibly file permission issue or running out of disk space.");